numpy/lib/utils.py

```diff
@@ -1029,14 +1029,12 @@ def _median_nancheck(data, result, axis, out):
     # masked NaN values are ok
     if np.ma.isMaskedArray(n):
         n = n.filled(False)
-    if result.ndim == 0:
-        if n == True:
-            if out is not None:
-                out[...] = data.dtype.type(np.nan)
-                result = out
-            else:
-                result = data.dtype.type(np.nan)
-    elif np.count_nonzero(n.ravel()) > 0:
+    if np.count_nonzero(n.ravel()) > 0:
```

Comment:
This rewrites more than strictly needed, but for performance it seemed to make sense to start with only doing anything if there was an actual `nan`. Somewhat surprisingly, this stanza is quite fast even for the scalar case:
```
In [19]: %timeit np.count_nonzero(np.True_.ravel()) > 0
1.07 µs ± 2.14 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [21]: b = np.array(True)

In [22]: %timeit b.ndim == 0 and np.True_ == True
776 ns ± 6.61 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
What happened to `out`?

Reply:
I looked at the two places where this code is used, and one always has `result is out` if `out` is given (which was already assumed to be the case for when `result` is an array...). 

But you are right that perhaps it is better to then just change the call sequence to remove `out` and change it accordingly where it is used. Shall I do that?

Reply:
Thanks for checking that. If `out` isn't used it would be best to remove it and document the assumptions about how the function should be used.

Reply:
OK, makes sense. I pushed a second MAINT commit that removes the `out` argument (and cleans up the call site in `function_base` a bit).